common/src/bitset.rs

```diff
@@ -196,6 +196,30 @@ impl BitSet {
         Ok(())
     }
 
+    pub fn deserialize(mut reader: impl Read) -> io::Result<Self> {
+        let mut max_buf = vec![0; 4];
+        reader.read_exact(&mut max_buf)?;
+        let max_value =
+            u32::from_le_bytes(max_buf.try_into().expect("Max value should be 4 bytes"));
```

Comment:
tantivy already uses the awesome byteorder crate, which lets you read numerics out of a stream with a given endianness directly!
```rust
use byteorder::{LittleEndian, ReadBytesExt};

let max_value = reader.read_u32::<LittleEndian>()?;
```